### PR TITLE
Optimize SAXStream

### PR DESCRIPTION
**Pull request by alubbe**
This PR brings over improvements and speed-ups from https://github.com/exceljs/exceljs/pull/1135 that work on nodejs v8. It will help make the diff of the other PR smaller

### CODE REVIEW

**Comment by alubbe** [comment]
Could we try to review/merge this PR? If anything here is unclear, is there anything I can improve?

**Comment by Siemienik** [comment]
@alubbe I'm going to review when I will have some time

File: lib/utils/sax-stream.js [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
-const saxes = require('saxes');
+const Stream = require('readable-stream');
+const {SaxesParser} = require('saxes');
 
-module.exports = class SAXStream extends Writable {
-  constructor() {
-    super();
-    this.sax = new saxes.SaxesParser();
+module.exports = class SAXStream extends Stream.Transform {
+  constructor(eventTypes) {
```
Comment (Siemienik) [inline lib/utils/sax-stream.js]: I see that `eventTypes` everywhere has the same value, as `sax-stream.js` is an internal utility I think you should use default value here.

**Comment by Siemienik** [comment]
image
perfect! 😄

File: lib/utils/sax-stream.js (hunk 1 above)
Comment (alubbe) [reply lib/utils/sax-stream.js]: Good catch - fixed in https://github.com/exceljs/exceljs/pull/1135

**Pull request merged**